### ISSUE TITLE
Unable to import citdsepdeterminer

### ISSUE THREAD

**Issue by ghost** [z-dragonl/Causal-Learner#1]
Hello, 

Thank you for your extremely useful code!  I was trying to run "example_GCS.m" and it couldn't run all the way through because of this error:  Thank you for your help!
Error using citdsepdeterminer
Error: File: citdsepdeterminer.m Line: 95 Column: 12
Unable to find or import 'citperformeddata'. Imported names must end with '.*' or be
fully qualified.

Error in MMHC_G2 (line 44)
CITDSepDeterminer = citdsepdeterminer(CITRCApplier, CITPValueEstimator);

Error in Causal_Learner (line 135)
        [DAG,time]=algorithm(data,alpha,ns,p,maxK);

Error in example_GCS (line 74)
[Result1,Result2]=Causal_Learner(alg_name,data,data_type,alpha);

**Comment by z-dragonl** [comment]
Hello，


Thank you very much for using our toolbox!


For the problem you found, if you didn't use to modify the parameters in "example_GCS.m" , I guess that such problem is caused by your Matlab version. You can use Matlab2016 or above version.


Please let me know if you still have any problems.



Regards,
Zhaolong



From: kollibylecoffee ***@***.***>
Date: 2021-08-03 03:53:56
To: z-dragonl/Causal-Learner ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [z-dragonl/Causal-Learner] Unable to import citdsepdeterminer (#1)

Hello,
Thank you for your extremely useful code! I was trying to run "example_GCS.m" and it couldn't run all the way through because of this error: Thank you for your help!
 Error using citdsepdeterminer
 Error: File: citdsepdeterminer.m Line: 95 Column: 12
 Unable to find or import 'citperformeddata'. Imported names must end with '.*' or be
 fully qualified.
Error in MMHC_G2 (line 44)
 CITDSepDeterminer = citdsepdeterminer(CITRCApplier, CITPValueEstimator);
Error in Causal_Learner (line 135)
 [DAG,time]=algorithm(data,alpha,ns,p,maxK);
Error in example_GCS (line 74)
 [Result1,Result2]=Causal_Learner(alg_name,data,data_type,alpha);
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ghost** [comment]
Thank you for your response!  I am using Matlab 2021a, but the error only comes up for alg  MMHC.  But I am able to use other algorithms like GSBN, PC and PC_stable just fine.  Thank you again for the extremely useful code.

**Comment by z-dragonl** [comment]
Hello,


Thank you for your reply!

Our package can run on Matlab2016a, and thus I'm very sorry I thought that the new version of Matlab will be compatible with the previous version.


For your question, you need to add ".*" after each imported name if you use Matlab2021a. 
For example:   import citperformeddata -->> import citperformeddata.*


Please let me know if you still have any problems.


Regards,
Zhaolong


发件人：kollibylecoffee ***@***.***>
发送日期：2021-08-03 23:19:21
收件人：z-dragonl/Causal-Learner ***@***.***>
抄送人：z-dragonl ***@***.***>,Comment ***@***.***>
主题：Re: [z-dragonl/Causal-Learner] Unable to import citdsepdeterminer (#1)

Thank you for your response! I am using Matlab 2021a, but the error only comes up for alg MMHC. But I am able to use other algorithms like GSBN, PC and PC_stable just fine. Thank you again for the extremely useful code.
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ghost** [comment]
Thanks again!  Yes, I had tried the ".*" before I posted my first comment (sorry I had explained that0, but still the same error.  But as I said, GSBN, PC, PCstable work very well, so thank you!

**Comment by z-dragonl** [comment]
OK，now you have 3 ways to run MMHC:

1. Use a lower version of Matlab, such as Matlab2016a.

2. Add ".*" after EACH name that needs to be imported if you use Matlab2021a, not just modify "citperformeddata".

3. Leave your email, I will send you a version that can run on Matlab2021a.
















At 2021-08-04 22:59:03, "kollibylecoffee" ***@***.***> wrote:

Thanks again! Yes, I had tried the ".*" before I posted my first comment (sorry I had explained that0, but still the same error. But as I said, GSBN, PC, PCstable work very well, so thank you!

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
